### PR TITLE
8312444: Delete unused parameters and variables in SocketPermission

### PR DESCRIPTION
**Pull request by Korov**
Removing unused parameter `defval` in `SocketPermission.initEphemeralPorts`, so the variable `PRIV_PORT_MAX` and `DEF_EPH_LOW` unused too.

Removing unused parameter `cname` in `SocketPermission.authorizedIPv4` and `SocketPermission.authorizedIPv6`.

Parameter 'action' must be not empty so `a.length > 0`, the check `a.length - 1 < 0` can be removed.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8312444](https://bugs.openjdk.org/browse/JDK-8312444): Delete unused parameters and variables in SocketPermission (**Enhancement** - P5)

### Reviewers
 * [Guoxiong Li](https://openjdk.org/census#gli) (@lgxbslgx - Committer) ⚠️ Review applies to [a07280d4](https://git.openjdk.org/jdk/pull/18086/files/a07280d4960218f2e884cd93462d31fea6db9ede)
 * [Andrey Turbanov](https://openjdk.org/census#aturbanov) (@turbanoff - Committer) ⚠️ Review applies to [9ebd6e12](https://git.openjdk.org/jdk/pull/18086/files/9ebd6e1279d7fc23ee1683070453ac1cd43f9868)
 * [Daniel Fuchs](https://openjdk.org/census#dfuchs) (@dfuch - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/18086/head:pull/18086` \
`$ git checkout pull/18086`

Update a local copy of the PR: \
`$ git checkout pull/18086` \
`$ git pull https://git.openjdk.org/jdk.git pull/18086/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 18086`

View PR using the GUI difftool: \
`$ git pr show -t 18086`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/18086.diff">https://git.openjdk.org/jdk/pull/18086.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/18086#issuecomment-1973619485)

### CODE REVIEW

**Comment by dfuch** [comment]
The proposed cleanup looks OK and reduces potential confusion - since the removed parameters where actually not used. However I wonder if we want to change this class at this time. I added the label `security` to get some input from security libs maintainers.

**Comment by Korov** [comment]
> The proposed cleanup looks OK and reduces potential confusion - since the removed parameters were actually not used. However I wonder if we want to change this class at this time. I added the label `security` to get some input from security libs maintainers.

Thanks @dfuch , If some modifications need to be deleted, I will delete them immediately.

**Review by lgxbslgx — APPROVED** [review]
The patch looks good. But the title of this issue may need to be adjusted, because you not only change the `SocketPermission.initEphemeralPorts` but also change other code. Or you can separate the unrelated code to another issue.

**Comment by Korov** [comment]
> The patch looks good. But the title of this issue may need to be adjusted, because you not only change the `SocketPermission.initEphemeralPorts` but also change other code. Or you can separate the unrelated code to another issue.

Hello @lgxbslgx , the title has been changed, Can you check if it's suitable?

**Comment by lgxbslgx** [comment]
> Hello @lgxbslgx , the title has been changed, Can you check if it's suitable?

Seems good. If nobody disagrees such mixed patch, I will help you change the title and content of the issue.

**Comment by Korov** [comment]
> > Hello @lgxbslgx , the title has been changed, Can you check if it's suitable?
> 
> Seems good. If nobody disagrees such mixed patch, I will help you change the title and content of the issue.

Thanks so much.

> > Hello @lgxbslgx , the title has been changed, Can you check if it's suitable?
> 
> Seems good. If nobody disagrees such mixed patch, I will help you change the title and content of the issue.

Thanks so much.

**Comment by Korov** [comment]
> > Hello @lgxbslgx , the title has been changed, Can you check if it's suitable?
> 
> Seems good. If nobody disagrees such mixed patch, I will help you change the title and content of the issue.

Thanks so much.

File: src/java.base/share/classes/java/net/SocketPermission.java
Comment (turbanoff) [inline src/java.base/share/classes/java/net/SocketPermission.java:707]: now `cname` parameter is unused in this method

**Comment by turbanoff** [comment]
Let's align declaration of `java.net.SocketPermission#NONE` field

**Comment by turbanoff** [comment]
Can we drop `java.net.SocketPermission#defaultDeny` variable? It seems it's always has `false` value (`setDeny` is never called)

File: src/java.base/share/classes/java/net/SocketPermission.java
Comment (Korov) [reply src/java.base/share/classes/java/net/SocketPermission.java]: `cname` has been deleted.

**Comment by Korov** [comment]
Hello @turbanoff , The code has been modified based on the review.

**Review by turbanoff — APPROVED** [review]

**Comment by lgxbslgx** [comment]
> Seems good. If nobody disagrees such mixed patch, I will help you change the title and content of the issue.

Have changed just now! And please wait for a formal reviewer to review this path.

**Comment by Korov** [comment]
> > Seems good. If nobody disagrees such mixed patch, I will help you change the title and content of the issue.
> 
> Have changed just now! And please wait for a formal reviewer to review this path.

Thanks so much!!!

File: src/java.base/share/classes/java/net/SocketPermission.java
Comment (dfuch) [inline src/java.base/share/classes/java/net/SocketPermission.java:622]: This is important for some internal whitebox test. Please revert this change.

**Review by dfuch — CHANGES_REQUESTED** [review]

**Review by dfuch — CHANGES_REQUESTED** [review]

File: src/java.base/share/classes/java/net/SocketPermission.java
Comment (Korov) [reply src/java.base/share/classes/java/net/SocketPermission.java]: The method had been reverted.

File: src/java.base/share/classes/java/net/SocketPermission.java
Comment (Korov) [reply src/java.base/share/classes/java/net/SocketPermission.java]: > This is important for some internal whitebox test. Please revert this change.

Reverted.

File: src/java.base/share/classes/java/net/SocketPermission.java
Comment (turbanoff) [reply src/java.base/share/classes/java/net/SocketPermission.java]: do you which test uses this method? I wasn't able to find it.

**Review by dfuch — APPROVED** [review]
@Korov I did a round of testing and results came back green. You're good to go. The SecurityManager has been deprecated for removal, so maybe it would be better to stay away from SecurityManager related classes from now on unless there some strong reason to make a change.

**Pull request closed without merging**
